core/templates/components/summary-tile/exploration-summary-tile.component.ts

```diff
@@ -139,20 +147,23 @@ export class ExplorationSummaryTileComponent implements OnInit, OnDestroy {
     this.windowRef.nativeWindow.location.href = this.getExplorationLink();
   }
 
-  getAverageRating(): number {
-    if (!this.getRatings) {
-      return null;
+  // Function will return undefined when Exploration Ratings are not present.
+  getAverageRating(): number | undefined {
+    if (this.getRatings) {
+      return this.ratingComputationService.computeAverageRating(
+        this.getRatings);
     }
-    return this.ratingComputationService.computeAverageRating(
-      this.getRatings);
+    return undefined;
   }
 
-  getLastUpdatedDatetime(): string {
-    if (!this.getLastUpdatedMsec) {
-      return null;
+  // Function will return null when the property 'getLastUpdatedMsec' is null
+  // or undefined.
```

Comment:
when can this property be undefined?

Reply:
`getLastUpdatedMsec` being an @Input property is not being initialized and only contains a value after component Interactions at run-time so we can't say when we receive this property.

Reply:
ah okay, `getLastUpdatedMsec` is not being initialized and this function is being called inside `ngOnInit` to set the property.